File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map=new Map();
        for(let i=0; i<nums.length;i++){
            map.set(nums[i],i);
        }
        for(let i=0;i<nums.length;i++){
            if(map.get(target-nums[i])){
                let x=map.get(target-nums[i]);
                    if(x!=i){
                    console.log(x);
                    return [x,i];
                    }
            }
        }
        return [];
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums=sorted(nums)
        ans=[]
        group={}
        for i,num in enumerate(nums):
            target=-num
            leftptr=i+1
            rightptr=len(nums)-1
            while leftptr<rightptr:
                sumcheck=nums[leftptr]+nums[rightptr]
                if sumcheck == target:
                    ans.append([num, nums[leftptr], nums[rightptr]])
                if sumcheck> target:
                    rightptr=rightptr-1
                else:
                    leftptr=leftptr+1
        unique=[]
        for x in ans:
            if x not in unique:
                unique.append(x)
        return unique

